typings/index.d.ts

```diff
@@ -114,47 +111,53 @@ interface OptionalExtraGetItemPropsOptions {
   [key: string]: any
 }
 
-export interface GetItemPropsOptions extends OptionalExtraGetItemPropsOptions {
+export interface GetItemPropsOptions<Item>
+  extends OptionalExtraGetItemPropsOptions {
   index?: number
-  item: any
+  item: Item
 }
 
-export interface PropGetters {
+export interface PropGetters<Item> {
   getRootProps: (options: GetRootPropsOptions) => any
   getButtonProps: (options?: GetButtonPropsOptions) => any
```

Comment:
Tried to add even more types, but it is hard to encode dynamic nature of this function

```ts
type GetItemPropsResult<Element> = {
  id: string,
} & (({
  onMouseMove: (e: React.MouseEvent<Element>) => void,
  onMouseDown: (e: React.MouseEvent<Element>) => void,
} & ({
  onPress: (e: React.MouseEvent<Element>) => void, // should be react native type
} | {
  onClick: (e: React.MouseEvent<Element>) => void,
})) | {})

getItemProps: <Element>({options, ...rest}: {options: GetItemPropsOptions<Item>}) => GetItemPropsResult<Element> & typeof rest
```